### ISSUE TITLE
add support for lion optimizer

### ISSUE THREAD

**Issue by faisalhr1997** [Linaqruf/kohya-trainer#90]
![image](https://user-images.githubusercontent.com/97628039/220545546-a56fbc61-8301-484f-a4dd-e54eee362bed.png)

**Comment by GeneralAwareness** [comment]
I added it and either I did something wrong or this thing is crap.

#1 Kohya has to add weights that it demands to be used.
#2 It ate all my 15 gigs of mem so I had to turn checkpointing back on.
#3 It took 30% longer for the same data I just used with adam.

I put in a ticket on the source, but no word yet.

**Comment by Linaqruf** [comment]
I added Lion and other Optimizer from latest kohya script, please try it
Different optimizer has different treatment for learning rate and loss target, so do at your own risk. 

Also it seems you can't use `DAdaptation` in colab free tier, because you need >15GB VRAM 
![image](https://user-images.githubusercontent.com/50163983/221347702-ef375e6f-ec77-456a-a85e-403e00096788.png)

**Comment by GeneralAwareness** [comment]
@Linaqruf Reports are coming in now that people are getting the same results I was where even a 24GB card is limited to BS3.  My BS3 on the T4 I had to turn back on the checkpointing so that was some of the slowdown but not all of it.

![image](https://user-images.githubusercontent.com/124339252/221348075-e7e3541e-4af4-4c9e-9d4b-4a7cee4ee8b6.png)

**Comment by Linaqruf** [comment]
well even the developer is still finds a way to make it better
![image](https://user-images.githubusercontent.com/50163983/221348284-6ad32d1b-db0f-4a6b-b35c-1d3e23880195.png)

https://github.com/lucidrains/lion-pytorch

I think Lion Optimizer is only effective for LLM training and not diffusion model.
